### PR TITLE
Added Tests

### PR DESCRIPTION
**Pull request by raysandeep**
Hey Samyak, 

I'm having an issue with the join room handler. 
The error is `[{"message":"must not be null","path":["getUser"]}]`
I've tried it on Graphql Playground getting the same error.
Do you mind helping me out with this?

### CODE REVIEW

**Comment by samyak-jain** [comment]
The tests aren't running for me. The reason for this is that the suite is not properly integrated. Basically, the environment is not setup for TestWebOAuthHandler. Instead of manually calling setup, you should leverage testify to do that for you. 
Use this as reference: https://github.com/samyak-jain/AgoraBackend/blob/d2aeecbb476ede00060f6a4d5ddbfc9151511af2/test/server_test.go

In my code, ```go test``` will not run this function https://github.com/samyak-jain/AgoraBackend/blob/d2aeecbb476ede00060f6a4d5ddbfc9151511af2/test/server_test.go#L136 by default. It will only run https://github.com/samyak-jain/AgoraBackend/blob/d2aeecbb476ede00060f6a4d5ddbfc9151511af2/test/server_test.go#L257

That particular test is what will bootstrap the testing of all the other functions. This is so that testify can manage them (i.e setup the environment correctly before the test)

You can notice the difference in my RoomCreationHandler, I'haven't call SetupTest since testify will automatically handle that. 

Your setup is not setting up the environment before the tests so it will not work. Try to follow the example I've linked so that your workflow looks like the following:

- [ ] All your tests should be part of the suite just like RoomCreationHandler in my example
- [ ] The tests should be bootstrapped from one function just like what I've done for TestGraphQLTestSuite

This should be a fairly simple and minor refactor. In fact, you can just make your changes over my unit-test branch and then make the pull request.

**Comment by raysandeep** [comment]
Got it. I'll update in few mins.

File: server_test.go [hunk 1]
```diff
@@ -181,19 +174,19 @@ func RoomCreationHandler(t *testing.T, bearerToken string) CreateChannel {
...
 	if err != nil {
 		t.Fatal("DB Connection Failed!")
 	}
 	var GraphQLTestSuite = GraphQLTestSuite{
-		DB:          database,
-		bearerToken: bearerTokenGlobal,
+		DB:    database,
+		Token: suite.Token,
 	}
 	GraphQLTestSuite.SetupSuite()
```
Comment (samyak-jain) [inline server_test.go]: You should be using the suite object passed as a parameter rather than creating a new GraphQLTestSuite

**Review by samyak-jain — CHANGES_REQUESTED** [review]

File: server_test.go (hunk 1 above)
Comment (raysandeep) [reply server_test.go]: Oh yeah shit. I'll change it

**Comment by raysandeep** [comment]
@samyak-jain I've added Database Testing as well. For the Join room handler there is no need for DB Testing coz we are not appending anything to DB.

File: server_test.go [hunk 2]
```diff
@@ -151,87 +141,59 @@ func RoomCreationHandler(method string, url string, t *testing.T, status int, be
...
+	c.MustPost(query, &decodedResponse, func(bd *client.Request) {
+		bd.Variables = map[string]interface{}{
+			"title":      "Hey",
+			"enablePSTN": true,
+		}
+	})
+	var channelData models.Channel
+	assert.Equal(t, decodedResponse.CreateChannel.Title, "Hey") //change this
+	assert.Equal(t, db.Where("hostpassphrase = ?", decodedResponse.CreateChannel.Passphrase.Host).First(&channelData).RecordNotFound(), false)
+	return decodedResponse
```
Comment (samyak-jain) [inline server_test.go:167]: 1. Try putting more data. Use Table Driven Development along with faker libraries to generate relevant and deterministic data. 
2. You are checking assertions for few of the variables. Check for all of them. For example, when enablePSTN is not true, the backend should not be generating any PSTN credentials. You need to test for that.

File: server_test.go [hunk 3]
```diff
@@ -151,87 +141,59 @@ func RoomCreationHandler(method string, url string, t *testing.T, status int, be
...
 }
 
-func TestRoomCreation(t *testing.T) {
-	url := "http://localhost:8080/query"
-	method := "POST"
-	createChannelDecoded = RoomCreationHandler(method, url, t, 401, bearerTokenGlobal+"wef")
-	createChannelDecoded = RoomCreationHandler(method, url, t, 200, bearerTokenGlobal)
+func (suite *GraphQLTestSuite) RoomCreation(t *testing.T) {
```
Comment (samyak-jain) [inline server_test.go:170]: There's no need to add t as a parameter for functions that are part of the test suite.

**Review by samyak-jain — CHANGES_REQUESTED** [review]

File: server_test.go (hunk 3 above)
Comment (raysandeep) [reply server_test.go:170]: Okay. I'll remove it. Probably used it to debug

File: server_test.go (hunk 2 above)
Comment (raysandeep) [reply server_test.go:167]: Okay got it

Comment (samyak-jain) [reply server_test.go:167]: Did you get a chance to look into this?

**Pull request merged**